src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeOcrValidator.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    private void handleValidationResponse(
+        ValidationResponse res,
+        Envelope envelope,
+        ScannableItem docWithOcr
+    ) {
+        log.info("Validation result for envelope {}: {}", envelope.getZipFileName(), res.status);
+        switch (res.status) {
+            case ERRORS:
+                String message = "OCR validation service returned OCR-specific errors. "
+                    + "Document control number: " + docWithOcr.getDocumentControlNumber() + ". "
+                    + "Envelope: " + envelope.getZipFileName() + ".";
+                throw new OcrValidationException(
+                    message,
+                    "OCR fields validation failed. Validation errors: " + res.errors
+                );
+            case WARNINGS:
+                log.info(
+                    "Validation ended with warnings. File name: {}",
+                    envelope.getZipFileName()
+                );
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void handleRestClientException(
+        Throwable exc,
+        String validationUrl,
+        Envelope envelope,
+        ScannableItem docWithOcr
+    ) {
+        log.error(
+            "Error calling validation endpoint. Url: {}, DCN: {}, doc type: {}, doc subtype: {}, envelope: {}",
+            validationUrl,
+            docWithOcr.getDocumentControlNumber(),
+            docWithOcr.getDocumentType(),
+            docWithOcr.getDocumentSubtype(),
+            envelope.getZipFileName(),
+            exc
+        );
+
+        if (exc instanceof NotFound) {
+            throw new OcrValidationException("Unrecognised document subtype " + docWithOcr.getDocumentSubtype());
+        }
+    }
+
+    private Optional<String> findValidationUrl(String poBox) {
+        Optional<String> validationUrl = containerMappings
+            .getMappings()
+            .stream()
+            .filter(mapping -> Objects.equals(mapping.getPoBox(), poBox))
+            .findFirst()
+            .map(ContainerMappings.Mapping::getOcrValidationUrl)
+            .filter(url -> !Strings.isNullOrEmpty(url));
+
+        if (validationUrl.isEmpty()) {
+            log.info("OCR validation URL for po box {} not configured. Skipping validation.", poBox);
+        }
+
+        return validationUrl;
+    }
+
+    private Optional<ScannableItem> findDocWithOcr(List<ScannableItem> docs) {
+        return docs
+            .stream()
+            .filter(it -> it.getOcrData() != null)
+            .findFirst();
+    }
+
+    private FormData toFormData(ScannableItem doc) {
+        return new FormData(
+            doc
+                .getOcrData()
+                .fields
+                .stream()
+                .map(it -> new OcrDataField(it.name.asText(), it.value.textValue()))
+                .collect(toList())
+        );
+    }
+
+    private String getFormType(ScannableItem item) {
+        return item.getDocumentSubtype();
+    }
```

Comment:
do we need this method?

Reply:
Also this is different from original
```
    private String getFormType(InputScannableItem item) {
        return item.documentType == InputDocumentType.SSCS1 ? "SSCS1" : item.documentSubtype;
    }
```
Can you explain?

Are there any other changes you've made?

Reply:
Because there is no `DocumentType.SSCS1`. In `EnvelopeMapper` class `InputDocumentType.SSCS1` is converted into `DocumentType.FORM` and `DocumentSubtype.SSCS1`.

Reply:
Looks like you are right because this is sent to the service. Will fix this.

Reply:
There are no other changes in `EnvelopeOcrValidator` compared to `OcrValidator`. But later on it will be changed to not throwing `OcrValidationException` in case of OCR validation error response from the service.